hail/python/test/hailtop/aiotools/test_copy.py

```diff
@@ -416,3 +417,34 @@ async def test_copy_src_parts(copy_test_context):
 
     await expect_file(fs, f'{dest_base}file1', 'src/a/file1')
     await expect_file(fs, f'{dest_base}subdir/file2', 'src/a/subdir/file2')
+
+
+@pytest.mark.asyncio
+async def test_file_and_directory_error_with_slash(router_filesystem):
+    sema, fs, bases = router_filesystem
+
+    src_base = await fresh_dir(fs, bases, 'gs')
+    dest_base = await fresh_dir(fs, bases, 'gs')
+
+    await fs.create(f'{src_base}a/empty/')
+    await fs.create(f'{src_base}a/foo')
```

Comment:
Sorry, my previous comments weren't clear.  Let me try again:

I'm imagining you have:

 - {src_base}/empty/ (which is an empty file) and {src_base}/empty/foo
 - {src_base}/non-empty/ (which is a non-empty file) and {src_base}/non-empty/bar
 - {src_base}/empty-only/  (which is empty, but empty-only has no other files in it)

Then copy empty, and verify that the empty directory file _wasn't_ copied, and foo was.

Then verify copying from `non-empty/` gets a FileAndDirectoryError.

Then copy empty-only/.  I think this should work (it is a directory).  If you're copying to the local filesystem, it should create the target directory (I suspect we're not doing that now).

```
$ mkdir scratch; cd scratch
$ mkdir a
$ cp -a a b
$ find .
.
./b
./a
```

Reply:
This is not working. The case when copying `{src_base}/empty/foo` results in FileNotFound as a Google Exception.

Reply:
Wait, sorry the Google Exception happens in the test's assert, not in the copy code.